client/mysqldump.c

```diff
@@ -6175,7 +6653,7 @@ int main(int argc, char **argv)
         dump_tablespaces_for_tables(*argv, (argv + 1), (argc - 1));
       dump_selected_tables(*argv, (argv + 1), (argc - 1));
     }
-    else
+    else if (argc > 0)
```

Comment:
how is that different than `else` ?
We cannot invoke empty dump.

Reply:
we can now `mysqldump --system=all`.